flycheck.el

```diff
@@ -5813,16 +5814,30 @@ https://github.com/rust-lang/rust/blob/master/src/libsyntax/json.rs#L67-L139"
          errors)))
 
     ;; Then we turn children messages into flycheck errors pointing to the
-    ;; location of the primary span.  According to the format, children
-    ;; may contain spans, but they do not seem to use them in practice.
+    ;; location of the primary span.
     (dolist (child children)
       (let-alist child
         (push
          (flycheck-error-new-at
-          primary-line
-          primary-column
+          ;; Use the line/column from the first span if there is one, or
+          ;; fallback to the line/column information from the primary span of
+          ;; the diagnostic.
+          (or (and (car .spans)
```

Comment:
I wonder if this `and` is needed: if `(car .spans)` is `nil`, then `assq` will return nil as well.
Hence maybe `(let-alist (car .spans) .column_start)` works?

Reply:
We can't nest `let-alist` without running into byte-compile warnings.  But we can certainly get rid of the `and` since `nil` is safe.

After programming in stricter languages than Elisp, you tend to put guards defensively ;)

Reply:
^^

Reply:
Which warnings do you get? I don't get any for the following in Emacs 24.5, nor on master:

```elisp
;;; test --- test
;;; Commentary:
;;; Code:

(require 'let-alist)

(defun ~/access (x)
  "X."
  (let-alist x
    (let-alist (car .x) .xx)))

(~/access '((x . (((xx . 1) (yy . 2)) ((xx . 4) (yy . 5))))))

(provide 'test)
;;; test.el ends here
```